calculus.tex

```diff
@@ -0,0 +1,81 @@
+\subsection{Derivatives}
+
+\newcommand{\ddx}{\frac{\partial}{\partial x}}
+\newcommand{\ddX}{\frac{\partial}{\partial X}}
+\newcommand{\ddy}{\frac{\partial}{\partial y}}
+
+Using matrix notation, taking derivatives with respect to vectors or matrices is tricky. With tensors, it's relatively easy \citep{laue+:2018}, and likewise with named tensors.
```

Comment:
I think it would be nice to discuss a bit more why this is an important issue that needs solving. The issues with matrix notation are really bad here :  General confusion over numerator / denominator notation. No ability to take matrix-matrix derivative without vectorizing. Transpose being important and very easy to get backward.

Reply:
29f5d179e9807aaff0407f11c19c7d47c6f60738